bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/forms2/service/impl/BahmniFormDetailsServiceImpl.java

```diff
@@ -57,6 +57,10 @@ public BahmniFormDetailsServiceImpl(PatientService patientService, VisitService
     public Collection<FormDetails> getFormDetails(String patientUuid, FormType formType, int numberOfVisits) {
         Patient patient = getPatient(patientUuid);
         List<Visit> visits = visitService.getVisitsByPatient(patient);
+
```

Comment:
Please add a comment on this method, to indicate what prompted us to write this. Something like: `//Warning: Not having this block results in ALL Encounters across ALL patients to be returned, which will bring down the system with an OutOfMemory. See JIRA ticket: BAH-3513.`

Reply:
Done @gsluthra